### PR TITLE
Vfx/equipment level vfx

### PR DESCRIPTION
**Pull request by hskim881028**
### Description
* Added weapon effects for each weapon enhancement section
enhancement 3 ~ 6 -> level 1 effect
enhancement 7 ~ 9 -> level 2 effect
enhancement 10 -> level 3 effect
* If you want to change the effect, just change the prefab in this script object.
aura!UNITO-UNDERSCORE!resource

### How to test
1. Overwrite the CreateAvatar script in that path : `..\Assets\_Scripts\Lib9c\lib9c\Lib9c\Action`
1. Make Genesis block in Unity Editor
1. Equip weapon

### Related Links
* [[이펙트] 무기 이펙트 속성별,단계별 적용 요청](https://app.asana.com/0/958521740385861/1200103900539258)
* [[이펙트] 무기별 강화 이펙트를 적용해주세요](https://app.asana.com/0/958521740385861/1200433675767323/f)

### Required Reviewers
@planetarium/9c-dev , Namyujeong , caesty

### Screenshot
image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1141562434100787/1200535469316695) by [Unito](https://www.unito.io)

### CODE REVIEW

File: nekoyume/Assets/_Scripts/Game/Character/PlayerSpineController.cs [hunk 1]
```diff
@@ -151,6 +154,22 @@ public void UpdateWeapon(Sprite sprite)
                 _clonedSkin.SetAttachment(_weaponSlotIndex, WeaponSlot, newWeapon);
             }
 
+            Destroy(_cachedWeaponVFX);
+
+            if (!(weaponVFXPrefab is null))
+            {
+                var parent = new GameObject(weaponId.ToString());
+                parent.AddComponent<BoneFollower>();
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/Game/Character/PlayerSpineController.cs:162]: Is it ok to add component in every update of weapon?

File: nekoyume/Assets/_Scripts/Game/Character/PlayerSpineController.cs [hunk 2]
```diff
@@ -151,6 +154,22 @@ public void UpdateWeapon(Sprite sprite)
...
+                parent.transform.SetParent(transform);
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/Game/Character/PlayerSpineController.cs:163]: Is it ok to add component on every update of weapon?

File: nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs [hunk 3]
```diff
@@ -41,5 +41,39 @@ public static GameObject GetCharacterTitle(int grade, string text)
             data.Title.GetComponentInChildren<TextMeshProUGUI>().text = text;
             return data.Title;
         }
+
+        public static GameObject GetAura(int id, int level)
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs:45]:
```suggestion
        public static GameObject GetWeaponAura(int id, int level)
```

File: nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs [hunk 4]
```diff
@@ -41,5 +41,39 @@ public static GameObject GetCharacterTitle(int grade, string text)
...
+            if (auras != null)
+            {
+                var index = WeaponLevelToIndex(level);
+                return index > -1 ? auras.prefab[index] : null;
+            }
+
+            return null;
+        }
+
+        private static int WeaponLevelToIndex(int level)
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs:59]: I think it is better to be exposed on `WeaponAuraScriptableObject`.

File: nekoyume/Assets/_Scripts/Game/Character/PlayerSpineController.cs (hunk 2 above)
Comment (hskim881028) [reply nekoyume/Assets/_Scripts/Game/Character/PlayerSpineController.cs:163]: The weapon effect cached before attaching the component is being destroyed.

**Review by boscohyun — APPROVED** [review]

**Review by Unengine — APPROVED** [review]

File: nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs [hunk 5]
```diff
@@ -41,5 +41,39 @@ public static GameObject GetCharacterTitle(int grade, string text)
...
+        {
+            if (3 <= level && level < 7)
```
Comment (Namyujeong) [inline nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs:61]: The first effect should be shown from the 4t, not the 3t.

**Review by Namyujeong — CHANGES_REQUESTED** [review]

File: nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs (hunk 5 above)
Comment (Namyujeong) [reply nekoyume/Assets/_Scripts/Helper/ResourcesHelper.cs:61]: 이거 빼고는 이펙트 멋있게 잘나오는거 다 확인했습니다!
-전투
- 캐릭터
- 캐릭터 선택

랭킹-타 유저 캐릭터 페이지에서는 못봤습니다.

**Comment by Namyujeong** [comment]
@hskim881028 코멘트 드린건 제가 그냥 수정했는데 혹시 모르니 내일 확인해보시고 머지해주세요~

**Pull request merged**